examples/webgl_loader_pcd.html

```diff
@@ -55,26 +55,40 @@
 				//scene.add( new THREE.AxesHelper( 1 ) );
 
 				const loader = new PCDLoader();
-				loader.load( './models/pcd/binary/Zaghetto.pcd', function ( points ) {
 
-					points.geometry.center();
-					points.geometry.rotateX( Math.PI );
-					points.name = 'Zaghetto.pcd';
-					scene.add( points );
+				const loadPointCloud = function ( file ) {
 
-					//
+					loader.load( './models/pcd/' + file, function ( points ) {
 
-					const gui = new GUI();
+						points.geometry.center();
+						points.geometry.rotateX( Math.PI );
+						points.name = file;
+						scene.add( points );
 
-					gui.add( points.material, 'size', 0.001, 0.01 ).onChange( render );
-					gui.addColor( points.material, 'color' ).onChange( render );
-					gui.open();
+						const gui = new GUI();
 
-					//
+						gui.add( points.material, 'size', 0.001, 0.01 ).onChange( render );
+						gui.addColor( points.material, 'color' ).onChange( render );
+						gui.add( points, 'name', [
+							'ascii/simple.pcd',
+							'binary/Zaghetto.pcd',
+							'binary/Zaghetto_8bit.pcd',
+							'binary_compressed/pcl_logo.pcd',
+						] ).name( 'type' ).onChange( e => {
 
-					render();
+							scene.remove( points );
```

Comment:
If you create a new GUI every time a new model is added, we should add the following before loading the new point cloud:
```js
gui.destroy();
```
This will make sure all DOM elements and event listeners associated with this GUI are destroyed.

Reply:
Get It